allow disabling profile logic in NodeGetListQuery

ajtmccarty opened the pull request:
adds a new env var `INFRAHUB_EXPERIMENTAL_NO_PROFILES` that removes the profile-related logic from `NodeGetListQuery` and `NodeManager.get_many`. this results in a substantially simpler `NodeGetListQuery` cypher query and an improvement in performance. my local testing showed perhaps a 10% improvement in performance when running a single query with and without the profiles logic, but I am not completely certain that there is actually a performance improvement. perhaps running one of the large generators the team has been working on is the best way to determine if this update actually improves performance

~this new env var also removes any profile-related mutations from GraphQL schema.~ this caused toast errors on the front end, so removed it

if a user has profiles linked to objects and then turns activates this setting, queries for objects linked to existing profiles will no longer be correct. I'm not sure if there are more guardrails that we should put in place for this

ajtmccarty committed:
allow disabling profile logic in NodeGetListQuery

ajtmccarty committed:
update unit test, add changelog

ajtmccarty committed:
add log statement

ajtmccarty committed:
turn off profile graphql mutations with new env var

dgarros reviewed (commented):
Would be good to validate if the frontend is able to work correctly when the profiles are disabled before merging this PR
if it doesn't work we might need to keep the GraphQL queries related to profiles for now

fatih-acar commented:
I get a rough 50% improvement on NodeGetList when running the following query:

```
query {
  NetworkPhysicalInterfaceL3(name__value: "Ethernet1/1/1", order: {disable: true}) {
    edges {
      node {
        id
      }
    }
  }
}
```

fatih-acar commented:
Not really the goal of this PR, but could we also do the same for NodeListGetInfoQuery? That would fully remove profiles overhead in the graphql query path.

ajtmccarty committed:
conditionally remove profiles from NodeManager.get_many

ajtmccarty committed:
fix var name

ajtmccarty committed:
remove whoopsidaisy

ajtmccarty committed:
get_many unit tests

ajtmccarty commented:
- removed the code the turned off profile-related mutations if the new `NO_PROFILES` setting was active
- updated `NodeManager.get_many`/`NodeListGetInfoQuery` to remove profile-related logic, depending on the new env var as well

fatih-acar commented:
After more thorough testing, I confirm what you've seen: we get a 10 to 20% improvement.
Still need to test it using the test framework with the generator.

ajtmccarty commented:
removing all profile's logic from read cypher queries in #7256
